app/models/piece.rb

```diff
@@ -6,6 +6,14 @@ class Piece < ApplicationRecord
   scope :are_captured, -> { where(is_captured: true) }
   scope :are_not_captured, -> { where(is_captured: false) }
 
+  def white?
+    self.user == self.game.white_player
```

Comment:
To be completely safe, can you change these checks to `self.user_id == self.game.white_player_id`?  I'm not 100% clear on when this happens, but sometimes this check will fail because the two user values are different model instances that both refer to the same database record.  So to be totally safe you can compare just the IDs.